tests/publishing/test_uploader.py

```diff
@@ -72,3 +72,31 @@ def test_uploader_registers_for_appropriate_400_errors(
         uploader.upload("https://foo.com")
 
     assert register.call_count == 1
+
+
+@pytest.mark.parametrize(
+    "status, body",
+    [
+        (409, ""),
+        (400, "already exist"),
+        (400, "updating asset"),
+        (403, "overwrite artifact"),
+        (400, "already been taken"),
```

Comment:
```suggestion
        (400, "File already exists"),
        (400, "Repository does not allow updating assets"),
        (403, "Not enough permissions to overwrite artifact"),
        (400, "file name has already been taken"),
```

The body should include more than the search string alone.